### PR TITLE
chore: update to countervalues V3 api

### PR DESCRIPTION
**Pull request by gre**
### 📝 Description

This PR migrates the countervalues client usage of `live-common/src/countervalues` from V2 to V3 backend API. This do not include the Market page section of the app but everything else.

- the `modules` logic has been reduced to remove unnecessary logic around the ethbtc rate inversion. (the API correctly allows to ask any `from->to` really)
- Tests has been reinforced to cover more cases.
- the pairs sorting has been moved from the api code directly into the logic part of the "inferTrackingPairForAccounts" logic.
- the "inferTrackingPairForAccounts" logic will also filter out the accounts that don't have any history yet.

#### changes for user land of the countervalues module

- in `TrackingPair` type, `startDate` becomes mandatory because the backend have required it to be mandatory since v3. Hopefully this did not impact a lot of parts (see Impact at the end of this PR description)
- remove unused `useCalculateMany`

#### What this did not solved, food for future improvements

- https://github.com/LedgerHQ/ledger-live/pull/5919
- the only remain module is "WETH" which may help to reduce one request to server, but this seems too overkill and specific to have this client side? Sure we know by contract proof that ETH=WETH, but there may be more than WETH to cover if we start thinking this way, it's questionable weither this logic should stay on client side, can be rediscussed in future

### ❓ Context

- **JIRA or GitHub link**: 
  - https://ledgerhq.atlassian.net/browse/LIVE-10623
  - https://ledgerhq.atlassian.net/browse/LIVE-10624
  - https://ledgerhq.atlassian.net/browse/LIVE-3513
  - https://ledgerhq.atlassian.net/browse/LIVE-10925 

### ✅ Checklist

Pull Requests must pass the CI and be code reviewed. Set as Draft if the PR is not ready.

- [x] `npx changeset` was attached.
- [x] **Covered by automatic tests.** 
- [x] **Impact of the changes:** 
  - countervalues – everywhere we see token and currency prices in fiat / crypto values – except the Market page
  - LLD and LLM display of countervalues.
  - Hedera fees estimation (it directly uses the api to calculate it)
  - "Troubleshooting" panel under Settings should still give a green check

---

### 🧐 Checklist for the PR Reviewers

- [ ] **The code aligns with the requirements** described in the linked JIRA or GitHub issue.
- [ ] **The PR description clearly documents the changes** made and explains any technical trade-offs or design decisions.
- [ ] **There are no undocumented trade-offs**, technical debt, or maintainability issues.
- [ ] **The PR has been tested** thoroughly, and any potential edge cases have been considered and handled.
- [ ] **Any new dependencies** have been justified and documented.
- [ ] **Performance** considerations have been taken into account. (changes have been profiled or benchmarked if necessary)

### CODE REVIEW

**Review by KVNLS — APPROVED** [review]
Great improvement, thanks! Small details in comments.

File: libs/ledger-live-common/src/countervalues/logic.ts [hunk 1]
```diff
@@ -488,13 +493,13 @@ function generateCache(
...
 
-  pairs.map(pair => {
+  for (const pair of pairs) {
     const { from, to } = resolveTrackingPair({
       from: pair.from,
       to: pair.to,
     });
 
-    if (from === to) return;
+    if (from === to) continue;
```
[the remark this answers is not in the document]
Comment (gre) [reply libs/ledger-live-common/src/countervalues/logic.ts:502]: ❓

[the remark this answers is not in the document]
Comment (gre) [reply libs/ledger-live-common/src/countervalues/logic.ts:502]: (moving from map to for loop, the code is equivalent)

**Comment by gre** [comment]
=> https://github.com/LedgerHQ/ledger-live/pull/5919

**Pull request closed without merging**